### ISSUE TITLE
OpenGL doesn't work when using RDP on Windows

### ISSUE THREAD

**Issue by GermanAizek** [Swartz27/xray-16#1528]
**Describe the bug**
Enviroment using RDP remote desktop. Game did not crash due to lack audiodevice, I created virtual audiodevice for OpenAL.

[openxray_r1.log](https://github.com/OpenXRay/xray-16/files/13365687/openxray_r1.log)
[openxray_r2a.log](https://github.com/OpenXRay/xray-16/files/13365683/openxray_r2a.log)
[openxray_r2.log](https://github.com/OpenXRay/xray-16/files/13365681/openxray_r2.log)
[openxray_r2.5.log](https://github.com/OpenXRay/xray-16/files/13365688/openxray_r2.5.log)
[openxray_r3.log](https://github.com/OpenXRay/xray-16/files/13365684/openxray_r3.log)
[openxray_r4.log](https://github.com/OpenXRay/xray-16/files/13365685/openxray_r4.log)
[openxray_rgl.log](https://github.com/OpenXRay/xray-16/files/13365686/openxray_rgl.log)

**Screenshots**
Screenshot render OpenGL using debugger.
![rgl](https://github.com/OpenXRay/xray-16/assets/21138600/e8d822ce-724f-4f4b-9b19-30369a988123)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - OpenXRay build version [[7dc0960](https://github.com/OpenXRay/xray-16/commit/8ff7b671369cdfbc44d203923966602f2bba10fc)]

**Comment by Xottab-DUTY** [comment]
https://www.khronos.org/opengl/wiki/Framebuffer_Object
https://registry.khronos.org/OpenGL-Refpages/gl4/html/glBindFramebuffer.xhtml

Your OpenGL drivers don't support OpenGL 4.1?

**Comment by GermanAizek** [comment]
No, it is very strange that r3 and r4 work perfectly.

**Comment by Xottab-DUTY** [comment]
> No

I assume that this answer means that your driver doesn't support OpenGL 4.1. I'm closing this then. Please, reopen if I'm wrong.

Seems like they just implemented DX11, but didn't paid much attention to OpenGL and DX9.

**Comment by GermanAizek** [comment]
> I assume that this answer means that your driver doesn't support OpenGL 4.1. I'm closing this then. Please, reopen if I'm wrong.
> 
> Seems like they just implemented DX11, but didn't paid much attention to OpenGL and DX9.

[strangely, everything should work, other games start without problems](https://dzen.ru/a/Y8WDVRTwwR8JZsVe)

**Comment by Xottab-DUTY** [comment]
> > I assume that this answer means that your driver doesn't support OpenGL 4.1. I'm closing this then. Please, reopen if I'm wrong.
> > Seems like they just implemented DX11, but didn't paid much attention to OpenGL and DX9.
> 
> [strangely, everything should work, other games start without problems](https://dzen.ru/a/Y8WDVRTwwR8JZsVe)

Other DX9 games start OK on your computer?

**Comment by GermanAizek** [comment]
dxdiag shows all feature levels
```
---------------
Display Devices
---------------
           Card name: Microsoft Remote Display Adapter
        Manufacturer: Microsoft
           Chip type: Unknown
            DAC type: Unknown
         Device Type: Display-Only Device
          Device Key: Enum\SWD\REMOTEDISPLAYENUM
       Device Status: 0180600A [DN_DRIVER_LOADED|DN_STARTED|DN_DISABLEABLE|DN_REMOVABLE|DN_NT_ENUMERATOR|DN_NT_DRIVER] 
 Device Problem Code: No Problem
 Driver Problem Code: Unknown
      Display Memory: 6144 MB
    Dedicated Memory: 0 MB
       Shared Memory: 6144 MB
        Current Mode: 1920 x 1080 (32 bit) (32Hz)
         HDR Support: Not Supported
    Display Topology: Internal
 Display Color Space: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
     Color Primaries: Red(0.000000,0.000000), Green(0.000000,0.000000), Blue(0.000000,0.000000), White Point(0.000000,0.000000)
   Display Luminance: Min Luminance = 0.000000, Max Luminance = 0.000000, MaxFullFrameLuminance = 0.000000
        Monitor Name: Generic Non-PnP Monitor
       Monitor Model: unknown
          Monitor Id: 
         Native Mode: unknown
         Output Type: Unknown
Monitor Capabilities: HDR Not Supported
Display Pixel Format: DISPLAYCONFIG_PIXELFORMAT_32BPP
      Advanced Color: Not Supported
         Driver Name: c:\windows\system32\drivers\umdf\rdpidd.dll,c:\windows\system32\drivers\wudfrd.sys
 Driver File Version: 10.00.19041.2075 (Russian)
      Driver Version: 10.0.19041.2075
         DDI Version: 12
      Feature Levels: 12_1,12_0,11_1,11_0,10_1,10_0,9_3,9_2,9_1
        Driver Model: WDDM 1.3
 Hardware Scheduling: Supported:False Enabled:False 
 Graphics Preemption: DMA
  Compute Preemption: DMA
            Miracast: Not Supported
      Detachable GPU: No
 Hybrid Graphics GPU: Not Supported
      Power P-states: Not Supported
      Virtualization: Not Supported
          Block List: No Blocks
  Catalog Attributes: N/A
   Driver Attributes: Final Retail
    Driver Date/Size: 21.06.2006 3:00:00, 254464 bytes
         WHQL Logo'd: Yes
     WHQL Date Stamp: Unknown
   Device Identifier: {D7B71AF4-43CC-11CF-5469-616AAFC2C735}
           Vendor ID: 0x1414
           Device ID: 0x008C
           SubSys ID: 0x00000000
         Revision ID: 0x0000
  Driver Strong Name: rdpidd.inf:c14ce88470d6efe0:RdpIdd_Install.NT:10.0.19041.2075:RdpIdd_IndirectDisplay
      Rank Of Driver: 00FF0000
         Video Accel: Unknown
         DXVA2 Modes: Unknown
      Deinterlace Caps: n/a
        D3D9 Overlay: Not Supported
             DXVA-HD: Not Supported
        DDraw Status: Not Available
          D3D Status: Enabled
          AGP Status: Not Available
       MPO MaxPlanes: 0
            MPO Caps: Not Supported
         MPO Stretch: Not Supported
     MPO Media Hints: Not Supported
         MPO Formats: Not Supported
    PanelFitter Caps: Not Supported
 PanelFitter Stretch: Not Supported
 ```
 
 I'll test the games later

**Comment by GermanAizek** [comment]
@Xottab-DUTY, synthetic tests DX9 all pass without problems

![image](https://github.com/OpenXRay/xray-16/assets/21138600/3fe26cb3-9dbd-4709-9173-20bc2da623a6)

![image](https://github.com/OpenXRay/xray-16/assets/21138600/7fb65ac0-3723-4753-a0d1-727de4559f2c)

**Comment by Xottab-DUTY** [comment]
> @Xottab-DUTY, synthetic tests DX9 all pass without problems
> 
> ![image](https://user-images.githubusercontent.com/21138600/284310082-3fe26cb3-9dbd-4709-9173-20bc2da623a6.png)
> 
> ![image](https://user-images.githubusercontent.com/21138600/284310309-7fb65ac0-3723-4753-a0d1-727de4559f2c.png)

Oh, oops! I actually forgot that this issue is about crash in OpenGL.
Do you have any synthetic test for OpenGL?

**Comment by Xottab-DUTY** [comment]
I also checked the log from DX9 renderers: the error you have is:
> E_FAIL ((0x80004005): Attempted to create a device with the debug layer enabled and the layer is not installed.

This is probably because d3dx9_31.dll is missing. This exact version of D3DX is being used to compile shaders.

**Comment by Xottab-DUTY** [comment]
I figured out that DX9 renderer don't work when d3dx9_31.dll is missing from user's computer. This was fixed (look in #1537).
OpenGL crash is separate thing, so I renamed this thread accordingly to make it specific to this OpenGL crash.

**Comment by GermanAizek** [comment]
@Xottab-DUTY maximum OpenGL 1.1 throught RDP connection its bug Windows NT systems. They implemented  only working with DirectX perfectly in RDP.
More info:
https://stackoverflow.com/questions/51705471/current-state-and-solutions-for-opengl-over-windows-remote

**Comment by Xottab-DUTY** [comment]
@GermanAizek, I just pushed a fix, that will make OpenGL renderer unavailable if `GLEW_ARB_framebuffer_object` is unsupported.
That should fix the issue when OpenGL is available in game options, but it doesn't work really. Please, check.

To fully fix this issue we would need to support OpenGL 1.1, but this is not possible right now.

**Comment by Xottab-DUTY** [comment]
> To fully fix this issue we would need to support OpenGL 1.1, but this is not possible right now.

Closing due to that. You can track for OpenGL 1.1 support (if it will be ever implemented) in #280.
